src/main/java/tigase/xmpp/impl/AbstractAuthPreprocessor.java

```diff
@@ -54,6 +60,14 @@
 			new ElementMatcher(Iq.IQ_QUERY_PATH, JabberIqAuth.ID, true)
 	};
 
+	protected boolean isBruteForceLockerEnabled(XMPPResourceConnection session) {
```

Comment:
I was thinking more about creating methods like `boolean isLoginAllowed(XMPPResourceConnection)` and `void reportInvalidLogin(session, clientIP, user_id)` and making BruteForceLockerBean field private as we could push all NPE checks in `AbstractAuthPreprocessor` class.

Reply:
Moved all members operating on brute force locker instance to `AbstractAuthPreprocessor`